### PR TITLE
Ensure version consistency between manifest and changelog

### PR DESCRIPTION
**Pull request by mtojek**
## What does this PR do?

This PR introduces additional validation to ensure consistency between manifest and changelog files.

## Why is it important?

This rule is required as developers tend to forget about updating changelog files.

## Checklist

- [x] I have added test packages to [`test/packages`](https://github.com/elastic/package-spec/tree/master/test/packages) that prove my change is effective.
- [x] I have added an entry in [`versions/N/changelog.yml`](https://github.com/elastic/package-spec/blob/master/versions/1/changelog.yml).

## Related issues

- Fixes https://github.com/elastic/package-spec/issues/167

### CODE REVIEW

**Comment by ycombinator** [comment]
I'm fine with the implementation the way it is, but if you want you can also use the [helpers from the `pkgpath` package](https://github.com/elastic/package-spec/blob/master/code/go/internal/pkgpath/files.go).

**Comment by ycombinator** [comment]
I'm not sure if the validation rule should be that the **first** entry in `changelog.yml` should match the version in the `manifest.yml`. I was thinking it should more be the **second** entry. 

Additionally the first entry should be a special version like `next` or something where package developers can accumulate multiple changes for the next version.

I proposed something like this in https://github.com/elastic/package-spec/issues/140#issue-814976510 as well.

**Comment by mtojek** [comment]
@ycombinator I transformed the source code to use `pkgpath` helpers and to check if the manifest version is referenced "somewhere" in the changelog.

File: code/go/internal/validator/semantic/version_integrity.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	f, err := pkgpath.Files(manifestPath)
+	if err != nil {
+		return "", errors.Wrap(err, "can't locate manifest file")
+	}
+
+	if len(f) != 1 {
+		return "", errors.New("single manifest file expected")
+	}
+
+	val, err := f[0].Values("version")
```
Comment (ycombinator) [inline code/go/internal/validator/semantic/version_integrity.go:43]: Doesn't this need to be `$.version`? Just `version` works?

**Review by ycombinator — APPROVED** [review]
Left a comment/question about `$.version` vs `version`. Otherwise, LGTM.

**Review by ycombinator — APPROVED** [review]
LGTM.

**Pull request merged**